### PR TITLE
Add support for a Jenkins version argument (CLI and env var) + Make Jenkins WAR parameter optional in the Plugin Manager Lib API and the implementation

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This PR adds two enhancements which allow Plugin Installation Manager to check plugin version compatibility when no Jenkins WAR supplied. It addresses use-cases like the jenkinsfile Runner of in https://github.com/jenkinsci/plugin-installation-manager-tool/issues/193#issuecomment-731418909 and https://github.com/jenkinsci/jenkinsfile-runner/issues/423

CC @timja @bericoberin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,9 @@ jenkins-plugin-cli --plugin-file /your/path/to/plugins.txt --plugins delivery-pi
 * `--plugin-file` or `-f`: (optional) Path to the plugins.txt, or plugins.yaml file, which contains a list of plugins to install. If this file does not exist, or if the file exists, but does not have a .txt or .yaml/.yml extension, then an error will be thrown. 
 * `--plugin-download-directory` or `-d`: (optional) Directory in which to install plugins. This configuration can also be made via the PLUGIN_DIR environment variable. The directory will be first deleted, then recreated. If no directory configuration is provided, the defaults are C:\ProgramData\Jenkins\Reference\Plugins if the detected operating system is Microsoft Windows, or /usr/share/jenkins/ref/plugins otherwise.
 * `--plugins` or `-p`: (optional) List of plugins to install (see plugin format below), separated by a space.
+* `--jenkinsVersion`: (optional) Version of Jenkins to be used.
```
Comment (timja) [inline README.md:37]: can we follow the format of the other vars please
```suggestion
* `--jenkins-version`: (optional) Version of Jenkins to be used.
```

File: plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java [hunk 2]
```diff
@@ -1056,15 +1060,36 @@ public String getAttributeFromManifest(File file, String key) {
...
+    /**
+     * Gets Jenkins version using one of the available methods.
+     * @return Jenkins version or {@code null} if it cannot be determined
+     */
+    @CheckForNull
+    public VersionNumber getJenkinsVersion() {
+        if (jenkinsVersion != null) {
+            return jenkinsVersion;
+        }
+        if (jenkinsWarFile != null) {
```
Comment (timja) [inline plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java:1072]: what do you think about moving this to the CLI? so the lib only works with a resolved jenkins version?

Comment (oleg-nenashev) [reply plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java:1072]: I do not think that Jenkins version is always needed for the library's logic. OTOH we could be more opinionated there

**Review by timja — APPROVED** [review]
:+1:

**Review by michaelneale — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Is it good to go now @timja, maybe towards 2.3.0?

**Comment by timja** [comment]
go for it.

**Comment by psaini0212** [comment]
Hello, @timja @oleg-nenashev the latest LTS release of Jenkins i.e. 2.263.4 still got v2.2.0 of plugin-installation-manager-tool. This is from the docker file -  ARG PLUGIN_CLI_URL=https://github.com/jenkinsci/plugin-installation-manager-tool/releases/download/2.2.0/jenkins-plugin-manager-2.2.0.jar
Do we know when the fixed/recent having the above fix will get baked in the Jenkins official image or is there a way I can bake it into that myself? pls guide

**Comment by timja** [comment]
create a new issue next time, it's being updated in https://github.com/jenkinsci/docker/pull/1073

**Pull request merged**